internal/controlplane/handlers_user.go

```diff
@@ -259,3 +262,132 @@ func (s *Server) GetUser(ctx context.Context, _ *pb.GetUserRequest) (*pb.GetUser
+	}
+
+	// Get the list of invitations for the user
+	invites, err := s.store.GetInvitationsByEmail(ctx, token.Email())
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return &pb.ListInvitationsResponse{}, nil
+		}
+		return nil, status.Errorf(codes.Internal, "failed to get invitations: %s", err)
+	}
+
+	// Build the response list of invitations
+	for _, invite := range invites {
+		// Get the sponsor's user information (current user)
+		sponsorUser, err := s.store.GetUserByID(ctx, invite.Sponsor)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get user: %s", err)
+		}
+		// Resolve the sponsor's identity and display name
+		identity := &auth.Identity{
+			Provider:  nil,
+			UserID:    sponsorUser.IdentitySubject,
+			HumanName: sponsorUser.IdentitySubject,
+		}
+		if flags.Bool(ctx, s.featureFlags, flags.IDPResolver) {
+			identity, err = s.idClient.Resolve(ctx, sponsorUser.IdentitySubject)
+			if err != nil {
+				zerolog.Ctx(ctx).Error().Err(err).Msg("error resolving identity")
+				return nil, util.UserVisibleError(codes.NotFound, "could not find identity %q", sponsorUser.IdentitySubject)
+			}
+		}
+		invitations = append(invitations, &pb.Invitation{
+			Code:           invite.Code,
+			Role:           invite.Role,
+			Email:          invite.Email,
+			Project:        invite.Project.String(),
+			CreatedAt:      timestamppb.New(invite.CreatedAt),
+			ExpiresAt:      timestamppb.New(invite.UpdatedAt.Add(7 * 24 * time.Hour)),
+			Expired:        time.Now().After(invite.UpdatedAt.Add(7 * 24 * time.Hour)),
+			Sponsor:        identity.UserID,
+			SponsorDisplay: identity.Human(),
+		})
+	}
+
+	return &pb.ListInvitationsResponse{
+		Invitations: invitations,
+	}, nil
+}
+
+// ResolveInvitation is a service for resolving an invitation.
+func (s *Server) ResolveInvitation(ctx context.Context, req *pb.ResolveInvitationRequest) (*pb.ResolveInvitationResponse, error) {
+	// Check if the UserManagement feature is enabled
+	if !flags.Bool(ctx, s.featureFlags, flags.UserManagement) {
+		return nil, status.Error(codes.Unimplemented, "feature not enabled")
+	}
+
+	// Extracts the user email from the token
+	tokenString, err := gauth.AuthFromMD(ctx, "bearer")
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no auth token: %v", err)
+	}
+	token, err := s.jwt.ParseAndValidate(tokenString)
+	if err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, "failed to parse bearer token: %v", err)
+	}
+
+	// Check if the invitation code is valid
+	ret, err := s.store.GetInvitationByCode(ctx, req.Code)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, util.UserVisibleError(codes.NotFound, "invitation not found")
+		}
+		return nil, status.Errorf(codes.Internal, "failed to get invitation: %s", err)
+	}
+
+	// Check if the invitation matches the user email
+	if token.Email() != ret.Email {
```

Comment:
Implemented this per - https://github.com/stacklok/minder/blob/07f93e0c1dcb59f74b6f15beb24cf684161287bb/database/query/invitations.sql#L15

which ensures if the email of the logged-in user corresponds to the one of the invite. 

@evankanderson - I know we discussed this to be more loose but then I saw that you made that comment so I implemented it that way. I'll ping you to see which is the desired behavior we want to keep. 

For the reviewers, don't take this comment as blocking as we can remove this check easily.